tests/test_algorithm.py

```diff
@@ -1818,20 +1821,21 @@ class TestClosePosAlgo(TestCase):
 
     def setUp(self):
         self.env = TradingEnvironment()
-        self.days = self.env.trading_days[:4]
+        self.days = self.env.trading_days[:5]
         self.panel = pd.Panel({1: pd.DataFrame({
             'price': [1, 1, 2, 4], 'volume': [1e9, 1e9, 1e9, 0],
             'type': [DATASOURCE_TYPE.TRADE,
                      DATASOURCE_TYPE.TRADE,
                      DATASOURCE_TYPE.TRADE,
                      DATASOURCE_TYPE.CLOSE_POSITION]},
-            index=self.days)
+            index=self.days[:4])
         })
         self.no_close_panel = pd.Panel({1: pd.DataFrame({
```

Comment:
As a result of auto-closing in `before_trading_start` as opposed to in `handle_market_close`, futures with an `auto_close_date` of the day after the last day of the simulation will no longer be auto-closed. With that said, I should work on a better test for that than this.